proto/model.proto

```diff
@@ -0,0 +1,151 @@
+syntax = "proto2";
+
+package com.signalfx.metrics.model;
+
+import "gogoproto/gogo.proto";
+
+option go_package = "model";
+
+// Enable gogoprotobuf extensions (https://github.com/gogo/protobuf/blob/master/extensions.md).
+// Enable custom Marshal method.
+option (gogoproto.marshaler_all) = true;
+// Enable custom Unmarshal method.
+option (gogoproto.unmarshaler_all) = true;
+// Enable custom Size method (Required by Marshal and Unmarshal).
+option (gogoproto.sizer_all) = true;
+// Enable registration with golang/protobuf for the grpc-gateway.
+option (gogoproto.goproto_registration) = true;
+
+enum MetricType {
+  /**
+   * Numerical: Periodic, instantaneous measurement of some state.
+   */
+  GAUGE = 0;
+  /**
+   * Numerical: Count of occurrences. Generally non-negative integers.
+   */
+  COUNTER = 1;
+  /**
+   * String: Used for non-continuous quantities (that is, measurements where there is a fixed
+   * set of meaningful values). This is essentially a special case of gauge.
+   */
+  ENUM = 2;
+  /**
+   * Tracks a value that increases over time, where only the difference is important.
+   */
+  CUMULATIVE_COUNTER = 3;
+}
+
+message Datum {
+  optional string strValue = 1;
+  optional double doubleValue = 2;
+  optional int64 intValue = 3;
+}
+
+message Dimension {
+  optional string key = 1 [
+    (gogoproto.nullable) = false
+  ];
+  optional string value = 2 [
+    (gogoproto.nullable) = false
+  ];
+}
+
+message DataPoint {
+  optional string source = 1 [
+    (gogoproto.nullable) = false
+  ];
+  optional string metric = 2 [
+    (gogoproto.nullable) = false
+  ];
+  optional int64 timestamp = 3 [
+    (gogoproto.nullable) = false
+  ];
+  optional Datum value = 4 [
+    (gogoproto.nullable) = false
+  ];
+  optional MetricType metricType = 5;
+  repeated Dimension dimensions = 6;
+}
+
+message DataPointUploadMessage {
+  repeated DataPoint datapoints = 1;
+}
+
+message PointValue {
```

Comment:
not sure if this is used :-/

Reply:
Me neither, I got it from the main proto. I would defer this call to others.

Reply:
agreed and it can be tracked separately also.